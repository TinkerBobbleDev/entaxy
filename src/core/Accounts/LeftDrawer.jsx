import React from 'react'
import { compose } from 'recompose'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import { withRouter } from 'react-router-dom'
import Drawer from '@material-ui/core/Drawer'
import List from '@material-ui/core/List'
import ListSubheader from '@material-ui/core/ListSubheader'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import Typography from '@material-ui/core/Typography'
import Tooltip from '@material-ui/core/Tooltip'
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction'
import IconButton from '@material-ui/core/IconButton'
import AddIcon from '@material-ui/icons/Add'
import EditIcon from '@material-ui/icons/Edit'
import SettingsIcon from '@material-ui/icons/Settings'
import grey from '@material-ui/core/colors/grey'
import InstitutionIcon from '../../common/InstitutionIcon'
import { currencyFormatter } from '../../util/stringFormatter'
import LinkTo from '../../common/LinkTo'

const styles = (theme) => ({
  root: {
    marginLeft: 200
  },
  drawerPaper: {
    width: 200,
    marginTop: 70,
    height: 'calc(100% - 70px)'
  },
  noAccounts: {
    background: grey[100],
    margin: '0 20px',
    padding: theme.spacing(1)
  },
  listSubheader: {
    background: '#f1f1f1',
    marginBottom: theme.spacing(1)
  },
  institutionListRoot: {
    padding: 0
  },
  institutionListItem: {
    paddingRight: 0
  },
  institutionListItemIcon: {
    minWidth: 30
  },
  institution: {
    padding: 0,
    textOverflow: 'ellipsis',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    maxWidth: 120,
    fontWeight: 500,
    display: 'inline-block',
    verticalAlign: 'bottom'
  },
  account: {
    padding: `0 0 0 ${theme.spacing(4)}px`
  },
  accountName: {
    fontSize: '95%',
    textOverflow: 'ellipsis',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    maxWidth: 110,
    display: 'inline-block',
    verticalAlign: 'middle'
  },
  ListItemText: {
    margin: '3px 0 3px 0'
  },
  smallFont: {
    fontSize: '95%',
    textOverflow: 'ellipsis'
  },
  smallButton: {
    padding: 4,
    marginRight: -6
  },
  smallIcon: {
    fontSize: 18
  },
  smallChipRoot: {
    fontSize: 9,
    height: 'auto',
    marginLeft: 10
  },
  smallChipLabel: {
    padding: '1px 7px'
  }
})

const mapStateToProps = (state) => ({
  settings: state.settings,
  accounts: state.accounts
})

export const LeftDrawerComponent = ({
  classes,
  settings,
  accounts,
  match,
  children
}) => {
  const userHasAccounts = Object.keys(accounts.byInstitution).length > 0
  const accountId = 'accountId' in match.params ? match.params.accountId : null
  return (
    <div className={classes.root}>
      <Drawer
        elevation={3}
        variant="permanent"
        classes={{ paper: classes.drawerPaper }}
      >
        <div className={classes.toolbar} />
        <List
          component="nav"
          dense={true}
          subheader={(
            <ListSubheader component="div" className={classes.listSubheader}>
              Accounts
              <ListItemSecondaryAction>
                <Tooltip id="tooltip-icon" title="New account">
                  <IconButton
                    aria-label="New account"
                    className={classes.smallButton}
                    component={LinkTo('/accounts/new')}
                  >
                    <AddIcon />
                  </IconButton>
                </Tooltip>
              </ListItemSecondaryAction>
            </ListSubheader>
          )}
        >
          {!userHasAccounts && (
            <Typography variant="caption" component="div" className={classes.noAccounts}>
              You don&apos;t have any accounts yet
            </Typography>
          )}
          {userHasAccounts && (
            <List dense={true} className={classes.institutionListRoot}>
              {Object.keys(accounts.byInstitution).sort().map((institution) => (
                Object.values(accounts.byInstitution[institution].groups).map((accountGroup) => (
                  <div key={`${institution}-${accountGroup.id}`}>
                    <ListItem>
                      <ListItemIcon className={classes.institutionListItemIcon}>
                        <InstitutionIcon institution={institution} size="small" />
                      </ListItemIcon>
                      <ListItemText
                        primary={institution}
                        className={classes.institutionListItem}
                        classes={{ primary: classes.institution }}
                        title={institution}
                      />
                      {accountGroup.type === 'api' && (
                        <ListItemSecondaryAction>
                          <Tooltip id="tooltip-icon" title="Edit API details">
                            <IconButton
                              aria-label="Edit API details"
                              className={classes.smallButton}
                              component={LinkTo(`/institutions/${institution}/import/${accountGroup.id}/edit`)}
                            >
                              <SettingsIcon className={classes.smallIcon} />
                            </IconButton>
                          </Tooltip>
                        </ListItemSecondaryAction>
                      )}
                    </ListItem>
                    <List dense={true} className={classes.institutionListRoot}>
                      {accountGroup.accountIds.map((id) => {
                        if (accounts.byId[id] === undefined) return undefined
                        const account = accounts.byId[id]
                        return (
                          <ListItem
                            className={classes.account}
                            button
                            key={account.id}
                            selected={account.id === accountId}
                            component={LinkTo(`/accounts/${account.id}/transactions`)}
                          >
                            <ListItemText
                              classes={{
                                root: classes.ListItemText,
                                primary: classes.accountName,
                                secondary: classes.smallFont
                              }}
                              primary={account.name}
                              secondary={
                                // eslint-disable-next-line max-len
                                currencyFormatter(settings.locale, account.currency)(account.currentBalance.accountCurrency)
                              }
                              title={account.name}
                            />
                            {accountGroup.type === 'default' && (
                              <ListItemSecondaryAction>
                                <Tooltip id="tooltip-icon" title="Edit account">
                                  <IconButton
                                    aria-label="Edit account"
                                    className={classes.smallButton}
                                    component={LinkTo(`/accounts/${id}/edit`)}
                                  >
                                    <EditIcon className={classes.smallIcon} />
                                  </IconButton>
                                </Tooltip>
                              </ListItemSecondaryAction>
                            )}
                          </ListItem>
                        )
                      })}
                    </List>
                  </div>
                ))
              ))}
            </List>
          )}
        </List>
      </Drawer>
      { children }
    </div>
  )
}

LeftDrawerComponent.propTypes = {
  classes: PropTypes.object.isRequired,
  settings: PropTypes.object.isRequired,
  accounts: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
  children: PropTypes.object.isRequired
}

export default compose(
  connect(mapStateToProps),
  withStyles(styles),
  withRouter
)(LeftDrawerComponent)
