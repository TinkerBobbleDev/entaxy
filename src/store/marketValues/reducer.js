import _ from 'lodash'
import types from './types'

export const initialState = {}

export default (state = initialState, action) => {
  switch (action.type) {
    case types.LOAD_MARKET_VALUES:
      return action.payload || initialState
    case types.CREATE_MARKET_VALUE:
      return {
        ...state,
        [action.payload.ticker]: {
          marketValue: action.payload.marketValue,
          updatedOn: action.payload.updatedOn
        }
      }
    case types.UPDATE_MARKET_VALUE:
      return {
        ...state,
        [action.payload.ticker]: {
          marketValue: action.payload.marketValue,
          updatedOn: action.payload.updatedOn
        }
      }
    case types.DELETE_MARKET_VALUES:
      return _.omit(state, action.payload)
    default:
      return state
  }
}
