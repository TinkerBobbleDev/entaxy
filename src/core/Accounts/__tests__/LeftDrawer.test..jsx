import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import { LeftDrawerComponent } from '../LeftDrawer'
import { initialState as accountsInitialState, groupByInstitution } from '../../../store/accounts/reducer'
import { initialState as settingsInitialState } from '../../../store/settings/reducer'

jest.mock('../../../common/InstitutionIcon', () => 'InstitutionIcon')

const accounts = {
  byId: {
    1: {
      id: '1',
      groupId: 0,
      description: 'Checking',
      institution: 'TD',
      currency: 'CAD',
      currentBalance: { accountCurrency: 1000, localCurrency: 1000 }
    },
    2: {
      id: '2',
      groupId: 0,
      description: 'Savings',
      institution: 'TD',
      currency: 'CAD',
      currentBalance: { accountCurrency: 2000, localCurrency: 2000 }
    },
    3: {
      id: '3',
      groupId: 0,
      description: 'Checking',
      institution: 'BMO',
      currency: 'USD', // Different currency
      currentBalance: { accountCurrency: 3000, localCurrency: 3000 }
    }
  },
  byInstitution: {
    Coinbase: {
      balance: 0,
      groups: {
        0: {
          id: 0,
          type: 'api',
          accountIds: [4]
        }
      }
    }
  }
}
accounts.byInstitution = groupByInstitution(accounts)

// Make sure we start with a specific currency
settingsInitialState.currency = 'XYZ'
settingsInitialState.locale = 'en-US'

describe('Accounts index (Left Nav)', () => {
  describe('snapshot', () => {
    it('matches with no accounts', () => {
      const component = renderer.create((
        <BrowserRouter>
          <LeftDrawerComponent
            settings={settingsInitialState}
            accounts={accountsInitialState}
            match={{ params: {} }}
            classes={{ }}
          >
            <div />
          </LeftDrawerComponent>
        </BrowserRouter>
      ))
      expect(component.toJSON()).toMatchSnapshot()
    })

    it('matches with a few accounts but none selected', () => {
      const component = renderer.create((
        <BrowserRouter>
          <LeftDrawerComponent
            settings={settingsInitialState}
            accounts={accounts}
            match={{ params: {} }}
            classes={{ }}
          >
            <div />
          </LeftDrawerComponent>
        </BrowserRouter>
      ))
      expect(component.toJSON()).toMatchSnapshot()
    })

    it('matches snapshot with a few account with one selected', () => {
      const component = renderer.create((
        <BrowserRouter>
          <LeftDrawerComponent
            settings={settingsInitialState}
            accounts={accounts}
            match={{ params: { accountId: 1 } }}
            classes={{ }}
          >
            <div />
          </LeftDrawerComponent>
        </BrowserRouter>
      ))
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
