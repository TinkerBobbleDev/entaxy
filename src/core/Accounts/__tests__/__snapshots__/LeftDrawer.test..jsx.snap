// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accounts index (Left Nav) snapshot matches snapshot with a few account with one selected 1`] = `
<div>
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div />
      <nav
        className="MuiList-root MuiList-dense MuiList-padding MuiList-subheader"
      >
        <div
          className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
        >
          Accounts
          <div
            className="MuiListItemSecondaryAction-root"
          >
            <a
              aria-describedby={null}
              aria-disabled={false}
              aria-label="New account"
              className="MuiButtonBase-root MuiIconButton-root"
              href="/accounts/new"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
              title="New account"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <ul
          className="MuiList-root MuiList-dense MuiList-padding"
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
              disabled={false}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <InstitutionIcon
                  institution="BMO"
                  size="small"
                />
              </div>
              <div
                className="MuiListItemText-root MuiListItemText-dense"
                title="BMO"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
                >
                  BMO
                </span>
              </div>
            </li>
            <ul
              className="MuiList-root MuiList-dense MuiList-padding"
            >
              <li
                className="MuiListItem-container"
              >
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
                  href="/accounts/3/transactions"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <div
                    className="MuiListItemText-root MuiListItemText-dense"
                  >
                    <p
                      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                    >
                      $3,000.00
                    </p>
                  </div>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
                <div
                  className="MuiListItemSecondaryAction-root"
                >
                  <a
                    aria-describedby={null}
                    aria-disabled={false}
                    aria-label="Edit account"
                    className="MuiButtonBase-root MuiIconButton-root"
                    href="/accounts/3/edit"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Edit account"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </a>
                </div>
              </li>
            </ul>
          </div>
          <div>
            <li
              className="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
              disabled={false}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <InstitutionIcon
                  institution="TD"
                  size="small"
                />
              </div>
              <div
                className="MuiListItemText-root MuiListItemText-dense"
                title="TD"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
                >
                  TD
                </span>
              </div>
            </li>
            <ul
              className="MuiList-root MuiList-dense MuiList-padding"
            >
              <li
                className="MuiListItem-container"
              >
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
                  href="/accounts/1/transactions"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <div
                    className="MuiListItemText-root MuiListItemText-dense"
                  >
                    <p
                      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                    >
                      CA$1,000.00
                    </p>
                  </div>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
                <div
                  className="MuiListItemSecondaryAction-root"
                >
                  <a
                    aria-describedby={null}
                    aria-disabled={false}
                    aria-label="Edit account"
                    className="MuiButtonBase-root MuiIconButton-root"
                    href="/accounts/1/edit"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Edit account"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </a>
                </div>
              </li>
              <li
                className="MuiListItem-container"
              >
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
                  href="/accounts/2/transactions"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <div
                    className="MuiListItemText-root MuiListItemText-dense"
                  >
                    <p
                      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                    >
                      CA$2,000.00
                    </p>
                  </div>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
                <div
                  className="MuiListItemSecondaryAction-root"
                >
                  <a
                    aria-describedby={null}
                    aria-disabled={false}
                    aria-label="Edit account"
                    className="MuiButtonBase-root MuiIconButton-root"
                    href="/accounts/2/edit"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Edit account"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </a>
                </div>
              </li>
            </ul>
          </div>
        </ul>
      </nav>
    </div>
  </div>
  <div />
</div>
`;

exports[`Accounts index (Left Nav) snapshot matches with a few accounts but none selected 1`] = `
<div>
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div />
      <nav
        className="MuiList-root MuiList-dense MuiList-padding MuiList-subheader"
      >
        <div
          className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
        >
          Accounts
          <div
            className="MuiListItemSecondaryAction-root"
          >
            <a
              aria-describedby={null}
              aria-disabled={false}
              aria-label="New account"
              className="MuiButtonBase-root MuiIconButton-root"
              href="/accounts/new"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
              title="New account"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <ul
          className="MuiList-root MuiList-dense MuiList-padding"
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
              disabled={false}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <InstitutionIcon
                  institution="BMO"
                  size="small"
                />
              </div>
              <div
                className="MuiListItemText-root MuiListItemText-dense"
                title="BMO"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
                >
                  BMO
                </span>
              </div>
            </li>
            <ul
              className="MuiList-root MuiList-dense MuiList-padding"
            >
              <li
                className="MuiListItem-container"
              >
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
                  href="/accounts/3/transactions"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <div
                    className="MuiListItemText-root MuiListItemText-dense"
                  >
                    <p
                      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                    >
                      $3,000.00
                    </p>
                  </div>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
                <div
                  className="MuiListItemSecondaryAction-root"
                >
                  <a
                    aria-describedby={null}
                    aria-disabled={false}
                    aria-label="Edit account"
                    className="MuiButtonBase-root MuiIconButton-root"
                    href="/accounts/3/edit"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Edit account"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </a>
                </div>
              </li>
            </ul>
          </div>
          <div>
            <li
              className="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
              disabled={false}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <InstitutionIcon
                  institution="TD"
                  size="small"
                />
              </div>
              <div
                className="MuiListItemText-root MuiListItemText-dense"
                title="TD"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
                >
                  TD
                </span>
              </div>
            </li>
            <ul
              className="MuiList-root MuiList-dense MuiList-padding"
            >
              <li
                className="MuiListItem-container"
              >
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
                  href="/accounts/1/transactions"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <div
                    className="MuiListItemText-root MuiListItemText-dense"
                  >
                    <p
                      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                    >
                      CA$1,000.00
                    </p>
                  </div>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
                <div
                  className="MuiListItemSecondaryAction-root"
                >
                  <a
                    aria-describedby={null}
                    aria-disabled={false}
                    aria-label="Edit account"
                    className="MuiButtonBase-root MuiIconButton-root"
                    href="/accounts/1/edit"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Edit account"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </a>
                </div>
              </li>
              <li
                className="MuiListItem-container"
              >
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
                  href="/accounts/2/transactions"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <div
                    className="MuiListItemText-root MuiListItemText-dense"
                  >
                    <p
                      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                    >
                      CA$2,000.00
                    </p>
                  </div>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
                <div
                  className="MuiListItemSecondaryAction-root"
                >
                  <a
                    aria-describedby={null}
                    aria-disabled={false}
                    aria-label="Edit account"
                    className="MuiButtonBase-root MuiIconButton-root"
                    href="/accounts/2/edit"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Edit account"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </a>
                </div>
              </li>
            </ul>
          </div>
        </ul>
      </nav>
    </div>
  </div>
  <div />
</div>
`;

exports[`Accounts index (Left Nav) snapshot matches with no accounts 1`] = `
<div>
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div />
      <nav
        className="MuiList-root MuiList-dense MuiList-padding MuiList-subheader"
      >
        <div
          className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
        >
          Accounts
          <div
            className="MuiListItemSecondaryAction-root"
          >
            <a
              aria-describedby={null}
              aria-disabled={false}
              aria-label="New account"
              className="MuiButtonBase-root MuiIconButton-root"
              href="/accounts/new"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
              title="New account"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          className="MuiTypography-root MuiTypography-caption"
        >
          You don't have any accounts yet
        </div>
      </nav>
    </div>
  </div>
  <div />
</div>
`;
